packages/amplify-category-function/src/provider-utils/awscloudformation/utils/lambda-layer-cloudformation-template.ts

```diff
@@ -0,0 +1,130 @@
+import { Fn, DeletionPolicy, Refs } from 'cloudform';
+import _ from 'lodash';
+import Lambda from 'cloudform-types/types/lambda';
+import { Permission, LayerParameters, getLayerMetadataFactory, LayerMetadata } from './layerParams';
+
+function generateLayerCfnObjBase() {
+  const cfnObj = {
+    AWSTemplateFormatVersion: '2010-09-09',
+    Description: 'Lambda Layer resource stack creation using Amplify CLI',
+    Parameters: {
+      layerVersion: {
+        Type: 'String',
+        Default: '1',
+      },
+      env: {
+        Type: 'String',
+      },
+    },
+    Resources: {},
+    Conditions: {
+      HasEnvironmentParameter: Fn.Not(Fn.Equals(Fn.Ref('env'), 'NONE')),
+    },
+  };
+  return cfnObj;
+}
+
+/**
+ * generates CFN for Layer and Layer permissions when updating layerVersion
+ */
+export function generateLayerCfnObj(context, parameters: LayerParameters) {
+  const layerData = getLayerMetadataFactory(context)(parameters.layerName);
+  const outputObj = {
+    Outputs: {
+      Arn: {
+        Value: Fn.Ref('LambdaLayer'),
+      },
+      Region: { Value: Refs.Region },
+    },
+  };
+  let cfnObj = { ...generateLayerCfnObjBase(), ...outputObj };
+  const POLICY_RETAIN = DeletionPolicy.Retain;
+  const latestVersion = layerData.getLatestVersion();
+  const layer = new Lambda.LayerVersion({
+    CompatibleRuntimes: parameters.runtimes.map(runtime => runtime.cloudTemplateValue),
+    Content: {
+      S3Bucket: Fn.Ref('deploymentBucketName'),
+      S3Key: Fn.Ref('s3Key'),
+    },
+    Description: `Lambda Layer version ${latestVersion}`,
+    LayerName: parameters.layerName,
+    LicenseInfo: 'MIT',
```

Comment:
Is this something we're comfortable with adding a license? If that is optional I don't think we should include it if legal did not clear it.

Reply:
This was defined by the design document, not something I have an opinion on. CC @brene